### PR TITLE
[FLINK-22795] Throw better exception when executing remote SQL file in SQL Client

### PR DESCRIPTION
**Pull request by JasonLeeCoding**
## What is the purpose of the change

*Throw better exception when executing remote SQL file in SQL Client*

## Brief change log

*add checkArgs method in SqlClient.java*

## Verifying this change

This change added tests and can be verified as follows:

*add testExecuteSqlWithHDFSFile in SqlClientTest.java*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by JasonLeeCoding** [comment]
@fsk119 Please help me review the code ,thanks.

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/SqlClientTest.java [hunk 1]
```diff
@@ -205,6 +205,17 @@ public void testExecuteSqlFile() throws Exception {
         }
     }
 
+    @Test
+    public void testExecuteSqlWithHDFSFile() throws Exception {
+        List<String> statements = Collections.singletonList("HELP;\n");
+        String sqlFilePath = createSqlFile(statements, "test-sql.sql");
+        String[] args = new String[] {"-f", "hdfs:/" + sqlFilePath};
```
Comment (fsk119) [inline flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/SqlClientTest.java:212]: Just use a invalid file path is enough. We don't need to create a new file here.

Should be `"hdfs://path/to/file"`?

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. Please always use rebase rather than merge in the Flink community. Please also revert the modification about `doc/themes/book`.

File: flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/SqlClientTest.java (hunk 1 above)
Comment (JasonLeeCoding) [reply flink-table/flink-sql-client/src/test/java/org/apache/flink/table/client/SqlClientTest.java:212]: yes, u are rigth just need a path

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlExecutionException.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.client;
+
+/** Exception thrown during the execution of SQL statements. */
+public class SqlExecutionException extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (JasonLeeCoding) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlExecutionException.java:22]: SqlClientException is currently under the gateway package. If it is not placed outside, the error message may mislead the user.  for example, org.apache.flink.table.client.gateway.SqlExecutionException: Currently, Flink doesn't support HDFS path. You can use local path.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java [hunk 3]
```diff
@@ -213,6 +214,16 @@ protected static void startClient(String[] args, Supplier<Terminal> terminalFact
         }
     }
 
+    public static void checkArgs(String[] args) {
+        for (String arg : args) {
+            if (arg.trim().toLowerCase().startsWith("hdfs")) {
```
[the remark this answers is not in the document]
Comment (JasonLeeCoding) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java:219]: OK, I will modify it to this way.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlExecutionException.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (JasonLeeCoding) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlExecutionException.java:22]: I'm sorry I misunderstood

**Pull request closed without merging**